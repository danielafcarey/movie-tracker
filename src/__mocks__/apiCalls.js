import { mockMovies } from '../mockData';

export const fetchMovies = jest.fn().mockImplementation( () => {
  return Promise.resolve( mockMovies );
});
