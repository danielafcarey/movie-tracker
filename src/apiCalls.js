import { apiKey } from './apiKey';

const fetchMovies = async () => {
  const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${apiKey}`;
  try {
    const response = await fetch(url);
    if (response.status === 200) {
      const data = await response.json();
      return data;
    } else {
      throw Error(`Status failure: ${response.status}`);
    }
  } catch (error) {
    throw error;
  }
}; 

const fetchUsers = async () => {
  const url = 'http://localhost:3000/api/users';
  try {
    const response = await fetch(url);
    if (response.status === 200) {
      const data = await response.json();
      return data.data;
    } else {
      throw Error(response.status);
    }
  } catch (error) {
    throw Error(error);
  }

};

const fetchFavorites = async (id) => {
  const url = `http://localhost:3000/api/users/${id}/favorites`;
  try {
    const response = await fetch(url);
    if (response.status === 200) {
      const data = await response.json();
      return data.data;
    } else {
      throw Error(response.status);
    }
  } catch (error) {
    throw Error(error);
  }
}

const postFavorite = async (userId, movie) => {
  const url = `http://localhost:3000/api/users/favorites/new`;
  const favorite = Object.assign({}, movie, { user_id: userId, overview: '' });
  const optionsObject = {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(favorite)
  };

  try {
    const response = await fetch(url, optionsObject);  
    if (response.status !== 200) {
      throw Error(response.status);
    }
  } catch (error) {
    throw Error(error);
  }
}

const deleteFavorite = async (userId, movieId) => {
  const url = `http://localhost:3000/api/users/${userId}/favorites/${movieId}`; 
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error(response.status)
    }
  } catch(error) {
    throw Error(error);
  }
}

export {
  fetchMovies,
  fetchUsers,
  fetchFavorites,
  postFavorite,
  deleteFavorite
};

