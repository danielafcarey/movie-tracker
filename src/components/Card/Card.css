.card {
  width: 200px;
  border: 1px black solid;
  display: flex;
  flex-direction: column; }
  .card img {
    width: 200px; }
  .card button {
    margin: 0 auto; }
