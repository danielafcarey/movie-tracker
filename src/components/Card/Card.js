import React from 'react';

const Card = (props) => {
  const {
    id,
    title,
    rating,
    image,
    favorite
  } = props;

  return(
    <div className='card'>
      <h1>{title}</h1>
      <img src={image} />
      <h3>Rating: {rating}</h3>
      <button>Favorite</button>
    </div>
  )
}

export default Card
