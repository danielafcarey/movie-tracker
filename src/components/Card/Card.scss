.card {
  width: 200px;
  border: 1px black solid;
  display: flex;
  flex-direction: column;

  img {
    width: 200px;
  }

  button {
    margin: 0 auto;
  }
}
