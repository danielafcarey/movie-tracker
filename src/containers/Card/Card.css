.card {
  width: 300px;
  display: flex;
  flex-direction: column;
  margin: 10px 20px;
  box-shadow: 3px 6px 16px 11px #888888;
  border-radius: 9px;
  position: relative;
  color: #535555; }
  .card h2 {
    word-wrap: break-word;
    font-size: 30px;
    margin: 10px 10px 5px 10px; }
  .card h3 {
    padding-left: 10px;
    margin: 10px 0;
    font-weight: 400; }
  .card img {
    width: 100%;
    border-radius: 9px 9px 0 0; }
  .card button {
    margin: 0 auto 33px;
    position: absolute;
    right: 10px;
    top: 10px;
    background: white;
    border: none;
    border-radius: 15px;
    color: #D46A6A;
    font-size: 19px;
    padding: 6px; }
    .card button:hover {
      cursor: pointer;
      background: #D46A6A;
      color: white; }
    .card button:focus {
      outline: none;
      border: none; }
  .card .favorite {
    background: #D46A6A;
    color: white; }
