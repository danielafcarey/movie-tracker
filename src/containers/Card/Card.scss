.card {
  width: 430px;
  border: 1px black solid;
  display: flex;
  flex-direction: column;
  border: 1px black solid;
  display: flex;
  flex-direction: column;
  margin: 10px 20px;
  box-shadow: 3px 6px 16px 11px #888888;

  h2 {
    padding: 10px;
    word-wrap: break-word;
    font-size: 30px;
  }
  h3 {
    padding: 10px;
  }

  img {
    width: 100%;
  }

  button {
    margin: 0 auto 33px;
  }
}