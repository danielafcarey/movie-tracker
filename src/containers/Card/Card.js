import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as apiCalls from '../../apiCalls';
import PropTypes from 'prop-types';
import { 
  addFavoriteToMovies, 
  deleteFavoriteFromMovies, 
  addFavoriteToFavorites, 
  deleteFavoriteFromFavorites 
} from '../../actions';

class Card extends Component {
  handleClick = () => {
    const movieToPost = {
      movie_id: this.props.movieId,
      title: this.props.title,
      vote_average: this.props.rating,
      poster_path: this.props.image,
      release_date: this.props.releaseDate,
    };
    const movieToStore = {
      movieId: this.props.movieId,
      title: this.props.title,
      rating: this.props.rating,
      image: this.props.image,
      releaseDate: this.props.releaseDate,
      favorite: this.props.favorite
    };

    if (!this.props.userId) {
     alert('Please sign in or create an account to add favorites');
     return;
    }
    
    if (this.props.favorite === false) {
      apiCalls.postFavorite(this.props.userId, movieToPost);
      this.props.addFavoriteToFavorites(movieToStore); 
      this.props.addFavoriteToMovies(movieToStore);
    } else {
      apiCalls.deleteFavorite(this.props.userId, this.props.movieId);
      this.props.deleteFavoriteFromFavorites(this.props.movieId);
      this.props.deleteFavoriteFromMovies(this.props.movieId);
    }
  };

  render() {
    const {
      title,
      rating,
      image,
      releaseDate,
      favorite
    } = this.props;

    let favoriteClass;
    favorite ? favoriteClass = 'favorite' : favoriteClass = '';

    return (
      <div className='card'>
        <img src={image} alt={`${title} poster`}/>
        <h2>{title}</h2>
        <h3>Rating: {rating}</h3>
        <h3>Release Date: {releaseDate}</h3>
        <button 
          className={ favoriteClass }
          onClick={ this.handleClick }>â˜†</button>
      </div>
    );
  };
};

const mapStateToProps = (state) => ({
  userId: state.currentUser  
});

const mapDispatchToProps = (dispatch) => ({
  addFavoriteToMovies: (favoriteMovie) => dispatch(addFavoriteToMovies(favoriteMovie)),
  deleteFavoriteFromMovies: (movieId) => dispatch(deleteFavoriteFromMovies(movieId)),
  addFavoriteToFavorites: (favoriteMovie) => dispatch(addFavoriteToFavorites(favoriteMovie)),
  deleteFavoriteFromFavorites: (movieId) => dispatch(deleteFavoriteFromFavorites(movieId))
});

Card.propTypes = {
  movieId: PropTypes.number,
  title: PropTypes.string,
  rating: PropTypes.number,
  releaseDate: PropTypes.string,
  image: PropTypes.string,
  favorite: PropTypes.bool,
  addFavoriteToMovies: PropTypes.func, 
  deleteFavoriteFromMovies: PropTypes.func,
  addFavoriteToFavorites: PropTypes.func,
  deleteFavoriteFromFavorites: PropTypes.func,
  userId: PropTypes.number
}

export {
  Card,
  mapStateToProps,
  mapDispatchToProps,
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);


