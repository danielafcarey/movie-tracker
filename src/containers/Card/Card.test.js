import Card from './Card'
import { shallow } from 'enzyme'
import React from 'react'
import * as apiCalls from '../../apiCalls';

describe('Card', () => {

  it('matches the snapshot', () => {
    const mockProps = {
      id: 2342342,
      title: 'The Mighty Gasbag',
      rating: 100,
      image: 'www.flatulence.com/image/123132.png',
      favorite: false 
    }
    const wrapper = shallow(<Card {...mockProps} />)

    expect(wrapper).toMatchSnapshot();
  })

  describe('handleClick', () => {

    it('calls apiCalls.postFavorite with the correct arguments if favorite is false', () => {
       
    })

    it('calls props.addFavorite with the correct arguments if favorite is false', () => {

    })

    it('calls apiCalls.removeFavorite with the correct arguments if favorite is true', () => {

    })

    it('calls props.removeFavorite with the correct arguments if favorite is true', () => {

    })
  })
})
