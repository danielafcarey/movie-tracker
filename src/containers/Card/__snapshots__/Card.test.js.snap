// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[Function]}
    deleteFavorite={[Function]}
    favorite={false}
    image="www.flatulence.com/image/123132.png"
    movieId={2342342}
    rating={100}
    releaseDate="tomorrow"
    title="The Mighty Gasbag"
    userId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="www.flatulence.com/image/123132.png"
/>,
        <h2>
          The Mighty Gasbag
</h2>,
        <h3>
          Rating: 
          100
</h3>,
        <h3>
          Release Date: 
          tomorrow
</h3>,
        <button
          onClick={[Function]}
>
          Favorite
</button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "www.flatulence.com/image/123132.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Mighty Gasbag",
        },
        "ref": null,
        "rendered": "The Mighty Gasbag",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Rating: ",
            100,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Rating: ",
          100,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Release Date: ",
            "tomorrow",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Release Date: ",
          "tomorrow",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="www.flatulence.com/image/123132.png"
/>,
          <h2>
            The Mighty Gasbag
</h2>,
          <h3>
            Rating: 
            100
</h3>,
          <h3>
            Release Date: 
            tomorrow
</h3>,
          <button
            onClick={[Function]}
>
            Favorite
</button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "www.flatulence.com/image/123132.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Mighty Gasbag",
          },
          "ref": null,
          "rendered": "The Mighty Gasbag",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Rating: ",
              100,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Rating: ",
            100,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Release Date: ",
              "tomorrow",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Release Date: ",
            "tomorrow",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
